tests/unit/test_strike_service.py

```diff
@@ -0,0 +1,141 @@
+import pytest
+from pytest_mock.plugin import MockerFixture
+from moddingway.database.models import Strike
+from moddingway.services import strike_service
+from moddingway import enums
+from typing import List
+import discord
+
+
+@pytest.mark.asyncio
+@pytest.mark.parametrize(
+    "previous_points,total_points,expected_punishment",
+    [
+        (0, 1, "Nothing"),
+        (1, 8, "11 day exile"),
+        (8, 10, "14 day exile"),
+        (10, 11, "14 day exile"),
+        (1, 9, "11 day exile"),
+        (9, 12, "14 day exile"),
+        (12, 13, "14 day exile"),
+        (10, 15, "Permanent ban"),
+    ],
+)
+async def test_apply_punisment(
+    previous_points: int,
+    total_points: int,
+    expected_punishment: str,
+    mocker: MockerFixture,
+    create_db_user,
+):
+    mocked_logging_embed = mocker.Mock()
+    mocked_user = mocker.Mock(roles=[enums.Role.VERIFIED])
+    mocked_db_user = create_db_user(get_strike_points=total_points)
+    res = await strike_service._apply_punishment(
```

Comment:
Looking at `_apply_punishment`, I think you should have some patching around `ban_service.ban_user` and `exile_service.exile_user`, and you should make sure that they are being called the expected number of times and with parameters that make sense. pytest has the `mocker.ANY` that will let you match only on the exile duration if you do `assert_called_once_with` for the call to exile service